src/DSE.Open/AsciiStringComparer.cs

```diff
@@ -37,25 +37,25 @@ private sealed class AsciiCharSequenceComparerCaseInsensitive : AsciiStringCompa
 
         public override bool Equals(AsciiString x, AsciiString y) => x.EqualsCaseInsensitive(y);
 
-        public override int GetHashCode([DisallowNull] AsciiString obj)
+        public override int GetHashCode(AsciiString obj)
         {
             AsciiChar[]? rented = null;
 
             var source = obj.Span;
 
             try
             {
-                Span<AsciiChar> buffer = source.Length > StackallocThresholds.MaxByteLength
-                    ? (rented = ArrayPool<AsciiChar>.Shared.Rent(StackallocThresholds.MaxByteLength))
-                    : stackalloc AsciiChar[source.Length];
+                Span<AsciiChar> buffer = source.Length <= StackallocThresholds.MaxByteLength
+                    ? stackalloc AsciiChar[source.Length]
+                    : rented = ArrayPool<AsciiChar>.Shared.Rent(source.Length);
 
                 for (var i = 0; i < source.Length; i++)
                 {
                     buffer[i] = source[i].ToUpper();
                 }
 
                 var c = new HashCode();
-                c.AddBytes(MemoryMarshal.Cast<AsciiChar, byte>(buffer));
+                c.AddBytes(ValuesMarshal.AsBytes(buffer[..source.Length]));
```

Comment:
We also weren't slicing. If these hash codes are persisted anywhere this could be a breaking change but I doubt they are.